src/index.ml

```diff
@@ -362,7 +348,7 @@ struct
     let generation =
       match log with None -> 0L | Some log -> IO.get_generation log.io
     in
-    let log_async_path = log_async_path root in
+    let log_async_path = Layout.log_async ~root in
     (* If we are in readonly mode, the log_async will be read during sync_log so
        there is no need to do it here. *)
     if (not readonly) && Sys.file_exists log_async_path then (
```

Comment:
you missed this one

Reply:
ah, you only want to use IO.exists in the index creation?

Reply:
Indeed, good catch. Should be fixed.

Reply:
No it should be everywhere